### ISSUE TITLE
.oneOf checked field not returning in matchedData

### ISSUE THREAD

**Issue by jplusje** [ctavan/express-validator#541]
After upgrading from 4.3.0 to 5.0.3 the code which validates a posted e-mail to be either "not posted", "empty" or "valid email" stopped working.

**(basic / stripped code example)**
```
validator.check.oneOf([
    validator.check.body('email').optional().isEmpty(),
    validator.check.body('email').optional().isEmail()
]);

const data = validator.filter.matchedData(req);
```

In 4.3.0 when posting an actual e-mail address it would be available within the matched data, in 5.0.3 it's missing. I've downgraded the version (for testing purposes) and the code started working again.

**Comment by gustavohenke** [comment]
That's because `oneOf` was fixed to return only data from the successful chain.
Because both of your chains point to the same field, it will never be returned in `matchedData`'s result.

You can work around this by simplifying your validation: `body('email').optional({ checkFalsy: true }).isEmail()`
See the docs [here](https://github.com/ctavan/express-validator#optionaloptions) for details on what `checkFalsy` option is doing.

**Comment by jefftmills** [comment]
I noticed this, too, when upgrading from 4.3.0 to 5.0.3 with a slightly different usage and work-around:
```
check.oneOf([
  check.param('id').isMongoId(),
  check.param('id').isInt()
]),
```
Your comment above says "fixed to return only data from the successful chain", but even when a valid Mongo id is provided (satisfying the first chain) it is **not** passed through to matchedData().

My work-around is:
```
check.param('id').custom(function(value) {
  return validator.isMongoId(value) || validator.isInt(value);
}),
```
My expectation was that if either a valid Mongo id or integer is provided then it should appear in the matchedData({onlyValidData: true}) result because it is valid.

**Comment by wakiem** [comment]
Also running into problems with this:

My code:
```
router.post("/", oneOf([
		checkSchema(logInToken),
		checkSchema(resetPasswordToken),
		checkSchema(verifyEmailToken)
	]), function(req, res) {
		console.log("Checked");
		const errors = validationResult(req);
		if (!errors.isEmpty()) {
			return res.status(422).json({ errors: errors.mapped() });
		}
		const tokenRequest = matchedData(req);
		...
	});
```

Schemas:
```
module.exports = {
	logInToken: {
		type: {
			exists: true,
			in: ["body"],
			equals: {
				options: "log-in"
			}
		},
		email: {
			exists: true,
			in: ["body"]
		},
		password: {
			exists: true,
			in: ["body"]
		}
	},
	resetPasswordToken: {
		type: {
			exists: true,
			in: ["body"],
			equals: {
				options: "reset-password"
			}
		},
		email: {
			exists: true,
			in: ["body"]
		}
	},
	verifyEmailToken: {
		type: {
			exists: true,
			in: ["body"],
			equals: {
				options: "verify-email"
			}
		},
		email: {
			exists: true,
			in: ["body"]
		}
	}
};
```

When posting this:
```
{
	"type": "verify-email",
	"email": "test@gmail.com"
}
```

Then:
`matchedData(req)`

Returns:
`{ email: 'test@gmail.com' }`

**Comment by gustavohenke** [comment]
Fixed by #559.

**Comment by gustavohenke** [comment]
🚢 v5.1.1 is out with the fix!
